ports/esp32/esp32_rmt.c

```diff
@@ -46,105 +50,101 @@
-}
-
-#else
+// Decide RMT usage in the machine_bitstream_high_low_rmt implementation.
+bool esp32_rmt_bitstream_enabled = true;
 
-// MicroPython runs on core 1, so we can call the RMT installer directly and its
-// interrupt handler will also run on core 1.
-esp_err_t rmt_driver_install_core1(uint8_t channel_id) {
-    return rmt_driver_install(channel_id, 0, 0);
+static bool IRAM_ATTR esp32_rmt_tx_trans_done(rmt_channel_handle_t channel, const rmt_tx_done_event_data_t *edata, void *user_ctx) {
+    esp32_rmt_obj_t *self = user_ctx;
+    self->tx_ongoing -= 1;
+    return false;
 }
 
-#endif // MP_TASK_COREID==0
-
 static mp_obj_t esp32_rmt_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
     static const mp_arg_t allowed_args[] = {
-        { MP_QSTR_id,        MP_ARG_REQUIRED | MP_ARG_INT, {.u_int = -1} },
+        { MP_QSTR_id,                          MP_ARG_INT, {.u_int = -1} },
         { MP_QSTR_pin,       MP_ARG_REQUIRED | MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = mp_const_none} },
-        { MP_QSTR_clock_div,                   MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = 8} }, // 100ns resolution
+        { MP_QSTR_resolution_hz,               MP_ARG_KW_ONLY | MP_ARG_INT, {.u_obj = mp_const_none} },
+        { MP_QSTR_clock_div,                   MP_ARG_KW_ONLY | MP_ARG_INT, {.u_obj = mp_const_none} },
         { MP_QSTR_idle_level,                  MP_ARG_KW_ONLY | MP_ARG_BOOL, {.u_bool = false} }, // low voltage
         { MP_QSTR_tx_carrier,                  MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = mp_const_none} }, // no carrier
+        { MP_QSTR_num_symbols,                 MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = SOC_RMT_MEM_WORDS_PER_CHANNEL} },
     };
     mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
     mp_arg_parse_all_kw_array(n_args, n_kw, all_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
-    mp_uint_t channel_id = args[0].u_int;
+    // RMT channel is an opaque struct in current RMT API and channel_id is a dummy parameter
+    // mp_uint_t channel_id = args[0].u_int;
     gpio_num_t pin_id = machine_pin_get_id(args[1].u_obj);
-    mp_uint_t clock_div = args[2].u_int;
-    mp_uint_t idle_level = args[3].u_bool;
-    mp_obj_t tx_carrier_obj = args[4].u_obj;
 
-    if (esp32_rmt_bitstream_channel_id >= 0 && channel_id == esp32_rmt_bitstream_channel_id) {
-        mp_raise_ValueError(MP_ERROR_TEXT("channel used by bitstream"));
+    uint32_t clock_freq;
+    check_esp_err(esp_clk_tree_src_get_freq_hz(RMT_CLK_SRC_DEFAULT, ESP_CLK_TREE_SRC_FREQ_PRECISION_CACHED, &clock_freq));
+
+    mp_uint_t resolution_hz;
+    if (args[2].u_obj != mp_const_none && args[3].u_obj != mp_const_none) {
+        mp_raise_ValueError(MP_ERROR_TEXT("resolution_hz and clock_div are mutually exclusive"));
+    } else if (args[2].u_obj == mp_const_none && args[3].u_obj == mp_const_none) {
+        // default value
+        resolution_hz = 10000000;
+    } else if (args[3].u_obj == mp_const_none) {
```

Comment:
Suggest writing this as `args[2].u_obj != mp_const_none` to make it a little clearer.

Same with the `clock_div` argument below.

Reply:
Fixed by https://github.com/micropython/micropython/pull/16293/commits/cbfa75ff8d05857e5cf830c21bc72f1ac75af6ea